In [38]:
import cirarra

In [39]:
importlib.reload(cirarra)
%load_ext autoreload
%autoreload 2

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


Check if it works with stored data:

In [155]:
c = cirarra.trng(seed=123,path='data/method2/20082021142022.npy',verbose=True)

Applying Elias RBE... 40.43% efficiency.


In [156]:
c.rand(5)

[0.1411764705882353,
 0.3568627450980392,
 0.7450980392156863,
 0.7450980392156863,
 0.7098039215686275]

Running `c.rand()` again should yield a different set of numbers. 

In [157]:
c.rand(5)

[0.4823529411764706,
 0.41568627450980394,
 0.1803921568627451,
 0.7019607843137254,
 0.43529411764705883]

Note how the set of numbers are different. Now check reproducibility.

In [158]:
c2 = cirarra.trng(seed=123,path='data/method2/20082021142022.npy',verbose=True)

Applying Elias RBE... 40.43% efficiency.


In [159]:
c2.rand(5)

[0.1411764705882353,
 0.3568627450980392,
 0.7450980392156863,
 0.7450980392156863,
 0.7098039215686275]

Note how the set of random numbers is the same as before; the seeding works. Now check if it works if we want to take a new sample. Recommended $k=1024$ but that will take around 30 minutes. We instead use $k=10$ just to check if it works.

In [178]:
c3 = cirarra.trng(k=10, verbose=True)

Sampling at 435 MHz and 40 gain... Sampling at 435 MHz, 40 gain and k = 10... n = 0 (0.0%). Time elapsed: 3.5e-05s
Sampling at 435 MHz, 40 gain and k = 10... n = 26215 (10.0%). Time elapsed: 0.252833s
Sampling at 435 MHz, 40 gain and k = 10... n = 52430 (20.0%). Time elapsed: 0.506957s
Sampling at 435 MHz, 40 gain and k = 10... n = 78645 (30.0%). Time elapsed: 0.758877s
Sampling at 435 MHz, 40 gain and k = 10... n = 104860 (40.0%). Time elapsed: 1.0112s
Sampling at 435 MHz, 40 gain and k = 10... n = 131075 (50.0%). Time elapsed: 1.26342s
Sampling at 435 MHz, 40 gain and k = 10... n = 157290 (60.0%). Time elapsed: 1.516096s
Sampling at 435 MHz, 40 gain and k = 10... n = 183505 (70.0%). Time elapsed: 1.768134s
Sampling at 435 MHz, 40 gain and k = 10... n = 209720 (80.0%). Time elapsed: 2.020308s
Sampling at 435 MHz, 40 gain and k = 10... n = 235935 (90.0%). Time elapsed: 2.272429s
Done.
Saved in data/2021-09-04 13:45:16.124996.npy. Please use this as path argument next time.
Applying Eli

In [179]:
c3.rand(10)

[0.10196078431372549,
 0.21568627450980393,
 0.7529411764705882,
 0.6784313725490196,
 0.7529411764705882,
 0.4392156862745098,
 0.9411764705882353,
 0.10196078431372549,
 0.10980392156862745,
 0.8627450980392157]

Looks like it's working. Next time we call the function, we can just use the `path='data/2021-09-04 13:45:16.124996.npy'` argument.